common/lib/xmodule/xmodule/partitions/partitions_service.py

```diff
@@ -34,10 +34,11 @@ def get_all_partitions_for_course(course, active_only=False):
         Returns:
             A List of UserPartitions associated with the course.
     """
-    all_partitions = course.user_partitions + _get_dynamic_partitions(course)
-    if active_only:
-        all_partitions = [partition for partition in all_partitions if partition.active]
-    return all_partitions
+    if course:
```

Comment:
Why is it necessary to have this `if course` check? course is a required parameter and I would expect the caller to not pass in a Falsey value for course.

Reply:
This and a few other additional checks you see as part of the PR is added due to failing CI test cases. The previous CI test runs were not catching them because all the test cases related to the schedules were being skipped due to [this](https://github.com/edx/edx-platform/blob/master/openedx/core/djangoapps/schedules/management/commands/tests/test_send_course_update.py#L34) check.